src/pretix/base/models/customers.py

```diff
@@ -59,6 +59,8 @@ class Customer(LoggedModel):
                               default=settings.LANGUAGE_CODE,
                               verbose_name=_('Language'))
     last_modified = models.DateTimeField(auto_now=True)
+    external_identifier = models.CharField(default="", max_length=255, verbose_name=_('External identifier'), blank=True)
+    notes = models.TextField(default="", verbose_name=_('Notes'), blank=True)
```

Comment:
I somehow could not get the migration to work without the `default=""` – I thought this was the default (no pun intended) when `blank=True` and not provided otherwise.

Reply:
Generally, you are right, it's no good to have two states for new data. However:

- Django will make it pretty impossible to have an empty string here, since if you enter an empty string into the form it will cause NULL to be saved

- NULL is semantically more correct for "there is no external identifier", so (a) it's nicer in the API (b) it would support e.g. future uniqueness constraints, if we'd say external identifiers must be unique (they probably should not be) then multiple "" values would be a problem, but multiple NULLs would not
 
- On some database (versions) adding a non-NULL-able column to a pre-existing table causes the entire table to be rewritten. If the table is large, this is a problem. For example, if we already have millions of customers in the pretix.eu production database, adding a non-nullable column would create a possibly **long** full downtime of pretix since the entire database goes down in locks! As a rule, we therefore avoid adding non-nullable columns to tables with large numbers of rows like the plague. (In this case, it would be probably fine since **recent** PostgreSQL versions can handle this quickly **if there is a default value**).

Reply:
Thanks for explaining the reasoning behind nullable columns. Makes total sense!